File: 26.py
max = 0
maxValue = 0


	
	
def getCircle(var):
	print(var)
	s = "0."
	n = 0
	rest = 1
	while True:
		print(s)
		save  = rest
		while rest < var:
			rest = rest * 10
	
		rest = rest%var
		n = n + 1
		if rest == 0:
			break
		if save == rest:
			break
	return n
		
			
	
for x in range(2,10):
	z = getCircle(x)
	if z > max:
		max = z
		maxValue = x
print(maxValue)
	

File: 26.java
import java.util.*;
class a26{
	public static void main(String[] args){
		Integer rest = 1;
		int max = 0;
		int maxValue = 0;
		for (int x = 2;x<1000;x++){
			Set<Integer> set = new HashSet<Integer>();
			set.add(1);
			int n = 0;
			rest = 1;
			while (true){
				while (rest < x){
					rest = rest * 10;
					}
				rest = rest%x;
				if (rest == 0 || set.contains(rest)){
					if (max < n){
						max = n;
						maxValue = x;
						}
					break;
					}
				set.add(rest);
				n = n +1;
				}
			}
	
		System.out.println(maxValue);
	}
}
